src/fidesops/service/connectors/saas_connector.py

```diff
@@ -288,17 +317,21 @@ def mask_data(
     ) -> int:
         """Execute a masking request. Return the number of rows that have been updated"""
         query_config = self.query_config(node)
+        if not query_config.masking_request:
+            raise Exception(
+                f"No masking request configured for {node.address.collection}"
+            )
```

Comment:
I still think we should raise an exception if nothing is configured so the user can take the appropriate steps to remedy.   They may not realize that they haven't defined an `update` request, or they may have defined a `delete` request and not set MASKING_STRICT to `false`.  If they don't want an update to be attempted at all, an easy way to remedy is to not set a primary key on the collection (this feature might not be obvious).

What are people's thoughts on this? Alternatively, we could just warn here and exit.

Reply:
I think we should raise an exception, but make it super clear what steps are to remedy. Maybe a clearer message would say something like "Either no masking request configured or no valid masking request for {node.address.collection}. Check that `MASKING_STRICT` env var is appropriately set."

I'm a user wouldn't want `update` at all, couldn't they simply not define an `update`, and instead define a `delete`?

Reply:
thanks for the improved exception.  Yes, if the user doesn't want `update`, they should define a `delete`, but they also need to define `MASKING_STRICT=false`